### PR TITLE
Add posibility to define options directly in DSN

### PR DESCRIPTION
**Pull request by cataragak90**
Enable the possibility to directly define options in the DSN. If options are specified in both the DSN and within the messenger.yaml file, the ones from the DSN will take precedence.

### CODE REVIEW

File: src/Transport/QueueTransportFactory.php [hunk 1]
```diff
@@ -26,4 +38,48 @@ public function supports(string $dsn, array $options): bool
...
+     * @return array
+     */
+    private function castOptionsToTheRightType(array $options): array
+    {
+        foreach ($options as $key => $value) {
+            if (!isset(self::OPTIONS_TYPE_MAPPING[$key])) {
+                continue;
+            }
+
+            $options[$key] = match (self::OPTIONS_TYPE_MAPPING[$key]) {
```
Comment (SpartakusMd) [inline src/Transport/QueueTransportFactory.php]: `match()` was introduced in PHP 8.0 but the package is compatible with PHP >= 7.3

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -39,6 +39,7 @@ framework:
                     time_to_live: <time_to_live_in_seconds>
                     results_limit: <how_many_messages_to_read>
 ```
+Options can be configured via the DSN or via the options key under the transport in ```messenger.yaml```
```
Comment (SpartakusMd) [inline README.md]: Please add a full example of `MESSENGER_TRANSPORT_DSN`

File: src/Transport/QueueTransportFactory.php (hunk 1 above)
Comment (cataragak90) [reply src/Transport/QueueTransportFactory.php]: done, it was replaced with "switch"

File: README.md (hunk 2 above)
Comment (cataragak90) [reply README.md]: done

**Review by SpartakusMd — APPROVED** [review]

**Pull request merged**